### PR TITLE
Support ldap groups

### PR DESCRIPTION
**Pull request by jnduwamu**
- Added variables for LDAP authentication
- Added a new scenario for testing LDAP

### CODE REVIEW

File: molecule/docker-auth/molecule.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        warehouse: "/opt/indexima/warehouse"
+        indexima_log_path: "/var/log/indexima"
+        indexima_hive_log_dir: "/tmp/indexima/hive"
+        indexima_history_dir: "/opt/indexima/oldhistory"
+        indexima_history_export: "/opt/indexima/newhistory"
+        ha: 0
+        ldap: 1
+        ldap_url: ${LDAP_URL}
+        auth_username: ${AUTH_USERNAME}
+        auth_password: ${AUTH_PASSWORD}
```
Comment (indexima-team) [inline molecule/docker-auth/molecule.yml:35]: Need consistent variable name: LDAP_USERNAME and LDAP_PASSWORD

File: molecule/docker-auth/molecule.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        ldap_user_dnpattern: ${LDAP_USER_DNPATTERN}
+        ldap_group_classkey: ${LDAP_GROUP_CLASSKEY}
+        ldap_group_dnpattern: ${LDAP_GROUP_DNPATTERN}
+        ldap_group_filter: ${LDAP_GROUP_FILTER}
+        ldap_group_membershipkey: ${LDAP_GROUP_MEMBERSHIPKEY}
```
Comment (indexima-team) [inline molecule/docker-auth/molecule.yml:40]: Don't put these in env variable. It is not sensible data and it is tested in verify.yml so it needs to be set explicitly

File: molecule/docker-auth/molecule.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        nodes: 1
+        cores: 1
+        ram: 600
+        partitions: 8
+        warehouse: "/opt/indexima/warehouse"
+        indexima_log_path: "/var/log/indexima"
+        indexima_hive_log_dir: "/tmp/indexima/hive"
+        indexima_history_dir: "/opt/indexima/oldhistory"
+        indexima_history_export: "/opt/indexima/newhistory"
+        ha: 0
```
Comment (indexima-team) [inline molecule/docker-auth/molecule.yml:31]: remove all these (except warehouse)

File: tasks/config_visualdoop.yaml [hunk 4]
```diff
@@ -16,3 +16,14 @@
...
+- name: Copy ldap.properties template to servers
+  template:
+    src: ldap.properties
+    dest: "{{ vd2_path }}/config/ldap.properties"
+    owner: "{{ service_user }}"
+    group: "{{ service_group }}"
+    mode: '0644'
+  when:
+    - is_master == 1
+    - "{{ ldap| bool }} == 'true'"
```
Comment (indexima-team) [inline tasks/config_visualdoop.yaml:29]: ```ldap |bool```
is enough

File: molecule/docker-auth/molecule.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    - destroy
+    - syntax
+    - create
+    - prepare
+    - converge
+    - idempotence 
+    - side_effect
+    - verify
+    - cleanup
+    - destroy
```
Comment (indexima-team) [inline molecule/docker-auth/molecule.yml:61]: This can be deleted

**Review by indexima-team — CHANGES_REQUESTED** [review]

File: molecule/docker-auth/molecule.yml (hunk 1 above)
Comment (indexima-team) [reply molecule/docker-auth/molecule.yml:35]: Nvm actually AUTH is ok for future tests

Comment (indexima-team) [reply molecule/docker-auth/molecule.yml:27]: up

Comment (jnduwamu) [reply molecule/docker-auth/molecule.yml:27]: You commented on this yesterday saying the name AUTH is actually ok for future tests

**Review by indexima-team — APPROVED** [review]

**Pull request merged**